.circleci/config.yml

```diff
@@ -0,0 +1,44 @@
+aliases:
+  - &restore_sbt_cache
+    restore_cache:
+      key: sbt-{{ checksum "/tmp/scala_version" }}-cache
+
+  - &save_sbt_cache
+    save_cache:
+      key: sbt-{{ checksum "/tmp/scala_version" }}-cache-{{ epoch }}
+      paths:
+        - "~/.ivy2/cache"
+        - "~/.sbt"
+        - "~/cache/coursier"
+
+  - &build_steps
+    - checkout
+    - run: echo "${SCALA_VERSION}" > /tmp/scala_version
```

Comment:
To avoid race conditions between the two builds, we'd like to have a separate cache per Scala version.

See: https://discuss.circleci.com/t/cannot-use-circle-yml-environment-variables-in-cache-keys/10994